### PR TITLE
Move helper scripts and templates to a directory

### PR DESCRIPTION
**Pull request by yosifkit**

### CODE REVIEW

**Comment by Moghedrin** [comment]
LGTM :+1:

**Comment by yosifkit** [comment]
@tianon, did you want me to undo the `generate-repo-stub-readme.sh` move?  Any other suggestions?

File: update.sh [hunk 1]
```diff
@@ -2,6 +2,7 @@
 set -e
 
 cd "$(dirname "$(readlink -f "$BASH_SOURCE")")"
+helperDir='./.template-helpers'
```
Comment (tianon) [inline update.sh]: Is the `./` here necessary?

File: update.sh [hunk 2]
```diff
@@ -92,10 +93,10 @@ for repo in "${repos[@]}"; do
 			logo="![logo](https://raw.githubusercontent.com/docker-library/docs/master/$repo/logo.png)"
 		fi
 		
-		cp -v README-template.md "$repo/README.md"
+		cp -v "$helperDir/template.md" "$repo/README.md"
 		
-		echo '  TAGS => ./generate-dockerfile-links-partial.sh'
-		replace_field "$repo" 'TAGS' "$(./generate-dockerfile-links-partial.sh "$repo")"
+		echo "  TAGS => generate-dockerfile-links-partial.sh"
```
Comment (tianon) [inline update.sh]: This line can still be single quotes, right?

Comment (yosifkit) [inline update.sh]: My bad, I was going to put the full path and decided that was ugly output.

**Comment by yosifkit** [comment]
@tianon, updated!

**Comment by tianon** [comment]
Yeah, I think having `generate-repo-stub-readme.sh` at the top level makes sense since it's not really a template helper as much as a helper for us to use on other repos.

File: README.md [hunk 3]
```diff
@@ -29,15 +29,15 @@ This is the main script used to generate the `README.md` files for each image. T
 
 This is used by us to push the actual content of the READMEs to the Docker Hub as special access is required to modify the Hub description contents.
 
-## `generate-dockerfile-links-partial.sh`
+## `.template-helpers/generate-dockerfile-links-partial.sh`
 
 This script is used by `update.sh` to create the "Supported tags and respective `Dockerfile` links" section of each generated `README.md` from the information in the [official-images `library/` manifests](https://github.com/docker-library/official-images/tree/master/library).
 
-## `generate-repo-stub-readme.sh`
+## `.template-helpers/generate-repo-stub-readme.sh`
```
Comment (tianon) [inline README.md]: :wink:

**Comment by tianon** [comment]
LGTM :+1:

File: README.md (hunk 3 above)
Comment (yosifkit) [inline README.md]: Forgot that...  Should I move it?

Comment (tianon) [inline README.md]: Yeah, probably best to move it up by `update.sh` since it's supposed to be "user-facing", so to speak.

**Comment by tianon** [comment]
Travis is hung up for some reason, but the tests ran successfully. :+1:

**Pull request merged**